en/getting-started/first-game/index.md

```diff
@@ -1,153 +1,470 @@
+So we could add a property to the script like the following:
 
-    > **Note**: the name of the script in __Cocos Creator 3.0__ is the name of the component. This name is case sensitive! If the capitalization of the component name is incorrect, the component cannot be used correctly by the name!
+```typescript
+// Whether to receive the jump command
+private _startJump: boolean = false;
+```
 
-#### Writing script code
+With a boolean variable like `_startJump`, we can mark whether the current character is in the jump or not, to help us distinguish between different branching logic in the `update` method. Because obviously, we don't need to move the character when no input is received.
 
-There are already some pre-set code blocks in the `PlayerController` script. Example:
+> Q: Why do we handle it in the `update` method?
+>
+> A: The `update` method will be invoked by the engine in a specific time interval. If the FPS of our game is 60 (60 renders per second), then the `update` method will be called 60 times per second. We can simulate realistic continuous behaviors by using `update` as much as possible.
 
-```ts
-import { _decorator, Component } from 'cc';
+So we can change the code as follows:
+
+```typescript
+import { _decorator, Component, input, Input, EventMouse } from 'cc';
 const { ccclass, property } = _decorator;
 
 @ccclass("PlayerController")
 export class PlayerController extends Component {
-    /* class member could be defined like this */
-    // dummy = '';
 
-    /* use `property` decorator if your want the member to be serializable */
-    // @property
-    // serializableDummy = 0;
+    // Whether to receive the jump command
+    private _startJump: boolean = false;
+    // The jump step count
+    private _jumpStep: number = 0;
 
     start () {
         // Your initialization goes here.
+        input.on(Input.EventType.MOUSE_UP, this.onMouseUp, this);
     }
 
-    // update (deltaTime: number) {
-    //     // Your update function goes here.
-    // }
+    onMouseUp(event: EventMouse) {
+       
+
+    }
+
+    update(dt: number): void {
+        if( this._startJump){ // Handle branching logic of jump
+
+        }
+    }
 }
 ```
 
-This code is the structure needed to write a __component__. Scripts with this structure are **Components in Cocos Creator**. They can be attached to nodes in a __Scene__ and provide various functionality for controlling nodes. For detailed information review the [Script](../../scripting/index.md) documentation.
+Good, here we have completed the main framework of our Player controller, so we just need to think about how to move the Player in the `update` method.
+
+In Newtonian mechanics, we have learned that if an object is to move with uniform velocity, then its position must be like the following:
+
+```math
+P_1 = P_0 + v*t
+```
+
+In the other words, we can add the current position of the object plus the velocity multiplied by the time, when we can get the new position.
```

Comment:
when we can get the new position -> then we can get the new position  or   when we want to get the new position.